--- node_modules/prismarine-viewer/viewer/lib/models.js	2026-09-28 08:18:36.596444204 +0000
+++ node_modules/prismarine-viewer/viewer/lib/models.js	2026-09-28 08:18:36.604444235 +0000
@@ -481,7 +481,7 @@
 
 function getModelVariants (block, blockStates) {
   // air, cave_air, void_air and so on...
-  if (block.name.includes('air')) return []
+  if (block.name === 'air' || block.name.endsWith('_air')) return []
   const state = blockStates[block.name] ?? blockStates.missing_texture
   if (!state) return []
   if (state.variants) {
